bin/reportdaily.py

```diff
@@ -56,6 +65,163 @@ class MissingSubCommand(ValueError):
+    """This function will create a config file where the user data is stored as a dict"""
+    # time to ask the user for his data
+
+    # name
+    print("Please enter your full name --> example: 'Max Musterman'")
+    name = ask_for_input(args.name, "Enter your Name: ")
+
+    # team
+    team = ask_for_input(args.team, "Enter your Team: ")
+
+    year = int(ask_for_input(
+        args.year, "In which year did your start your apprenticeship ?: "))
+    # time
+
+    today_date = date.today()
+    print(
+        f"You are ready to go.  Entries for this {today_date} can  bee added now with reportdaily add")
+
+    # create a config file
+    config = ConfigParser()
+    config["settings"] = {'name': name,
+                          'team': team, 'current_day': today_date, 'start_year': year}
+
+    # create a config file
+    os.makedirs(os.path.dirname(CONFIGPATH), exist_ok=True)
+    with open(CONFIGPATH, 'w') as user_config:
+        config.write(user_config)
+    print(f"The file was created at this path {CONFIGPATH}")
+
+
+def ask_for_input(var, message):
+    if var is None:
+        var = input(message)
+    return var
+
+
+def how_to_change_config(args, CONFIGPATH):
+    """User can change or overwrite the configs via direct input or console"""
+    if args.name is None and args.year is None and args.team is None and args.change:
+        user_input_change(args, CONFIGPATH)
+        result_value = 0
+    else:
+        namespace_config_change(args, CONFIGPATH)
+
+        result_value = 1
+    show_config(CONFIGPATH)
+    return result_value
+
+
+def namespace_config_change(args, CONFIGPATH):
+    """Input changes direct via the console, with just filling the namespace"""
+    # store all the args from namespace
+    name = args.name
+    team = args.team
+    year = args.year
+    change = args.change
+
+    # add config parser to file
+    config = ConfigParser()
+    config.read(CONFIGPATH)
+    if change is True:
+        # overwrite if namespace is filled
+        if name is not None:
+            config.set("settings", "name", name)
+        if team is not None:
+            config.set("settings", "team", team)
+        if year is not None:
+            config.set("settings", "start_year", year)
+
+        with open(CONFIGPATH, "w") as configfile:
+            config.write(configfile)
+        # show config to the user , so changes are visible to the user
+    print("namespace_config was selected")
+
+
+def user_input_change(args, CONFIGPATH):
+    # all user options
+    choice_table = {"Name": "t1", "Team": "t2", "Year": "t3"}
+    tmp_input = ''
+    overwrite_input = ' '
+    change_data = ''
+
+    # show and ask user what he wants to overwrite
+    print("""
+        "What do you want to change?"
+        Your options are
+        Name
+        Team
+        Year
+        """)
+
+    # check for right user input
+    while(True):
+        tmp_input = input("Name, Team, Year? ")
+        if tmp_input in choice_table:
```

Comment:
Hmn, what happens when the user enters a lowercase(!) `name`? Or enter `NAME`? That will not be found in the `choice_table`. As such, the if clause will fail.

To avoid such cases, use the string method `.capitalize()`:

```suggestion
        tmp_input = input("Name, Team, Year? ").capitalize()
        if tmp_input in choice_table:
```

Reply:
Oh Thanks, i knew there is a way , but didnt now about capitalize()
Thanks